src/Microsoft.Health.Dicom.SqlServer/Features/CustomTag/SqlCustomTagStore.cs

```diff
@@ -73,36 +75,41 @@ public async Task AddCustomTagsAsync(IEnumerable<CustomTagEntry> customTagEntrie
             }
         }
 
-        public async Task<IEnumerable<CustomTagEntry>> GetCustomTagsAsync(string path, CancellationToken cancellationToken = default)
+        public async Task<IReadOnlyList<CustomTagStoreEntry>> GetCustomTagsAsync(string path, CancellationToken cancellationToken = default)
         {
             if (_schemaInformation.Current < SchemaVersionConstants.SupportCustomTagSchemaVersion)
             {
                 throw new BadRequestException(DicomSqlServerResource.SchemaVersionNeedsToBeUpgraded);
             }
 
-            List<CustomTagEntry> results = new List<CustomTagEntry>();
+            List<CustomTagStoreEntry> results = new List<CustomTagStoreEntry>();
 
             using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken))
             using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
             {
-                V2.GetCustomTag.PopulateCommand(sqlCommandWrapper, path);
+                VLatest.GetCustomTag.PopulateCommand(sqlCommandWrapper, path);
 
+                var executionTimeWatch = Stopwatch.StartNew();
                 using (var reader = await sqlCommandWrapper.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
                 {
                     while (await reader.ReadAsync(cancellationToken))
                     {
-                        (string tagPath, string tagVR, int tagLevel, int tagStatus) = reader.ReadRow(
-                           V2.CustomTag.TagPath,
-                           V2.CustomTag.TagVR,
-                           V2.CustomTag.TagLevel,
-                           V2.CustomTag.TagStatus);
-
-                        results.Add(new CustomTagEntry { Path = tagPath, VR = tagVR, Level = (CustomTagLevel)tagLevel, Status = (CustomTagStatus)tagStatus });
+                        (long tagKey, string tagPath, string tagVR, int tagLevel, int tagStatus) = reader.ReadRow(
+                            VLatest.CustomTag.TagKey,
+                            VLatest.CustomTag.TagPath,
+                            VLatest.CustomTag.TagVR,
+                            VLatest.CustomTag.TagLevel,
+                            VLatest.CustomTag.TagStatus);
+
+                        results.Add(new CustomTagStoreEntry(tagKey, tagPath, tagVR, (CustomTagLevel)tagLevel, (CustomTagStatus)tagStatus));
                     }
+
+                    executionTimeWatch.Stop();
+                    _logger.LogInformation(executionTimeWatch.ElapsedMilliseconds.ToString());
                 }
             }
 
-            return results;
+            return results.AsReadOnly();
```

Comment:
This isn't necessary unless you want to make doubly-sure that the caller isn't going to cast the result to a `List<T>` and mutate it. Instead, you can simply return `results`